pkg/protocol/engine/blocks/block.go

```diff
@@ -467,6 +482,89 @@ func (b *Block) SetAccepted() (wasUpdated bool) {
 	return wasUpdated
 }
 
+// IsScheduled returns true if the Block was scheduled.
+func (b *Block) IsScheduled() bool {
+	b.mutex.RLock()
+	defer b.mutex.RUnlock()
+
+	return b.scheduled
+}
+
+// SetScheduled sets the Block as scheduled.
+func (b *Block) SetScheduled() (wasUpdated bool) {
+	b.mutex.Lock()
+	defer b.mutex.Unlock()
+
+	if wasUpdated = !b.scheduled; wasUpdated {
```

Comment:
I am not sure I understand the question. You think we shoudl update b.enqueued when we set skipped as well?

Reply:
Imagine a scenario in which goroutine A calls `SetSkipped`, while goroutine B calls `SetDropped` on the same block. Both will successfuly set the value, but if we have the following check, only the first goroutine will update the flag. 

```suggestion
	if wasUpdated = !b.scheduled; wasUpdated && b.enqueued {
```

The question is if it's even possible that different goroutines set those fields in parallel? If yes, then I think it would be worth adding this check. Wdyt?

Reply:
Ok, I see what you mean, I added this check tothe set skipped, scheduled and dropped methods to be on the safe side.